systems/framework/lcm_driven_loop.h

```diff
@@ -198,14 +196,41 @@ class LcmDrivenLoop {
                     << simulator_->get_context().get_time()
                     << ", but stepping to " << time << std::endl;
           std::cout << "Difference is too large, resetting " + diagram_name_ +
-              " time.\n";
+                           " time.\n";
           simulator_->get_mutable_context().SetTime(time);
         }
 
         simulator_->AdvanceTo(time);
         // Force-publish via the diagram
         diagram_ptr_->Publish(diagram_context);
+
+        // Clear messages in the current input channel
+        name_to_input_sub_map_.at(active_channel_).clear();
+      }
+
+      // Update the name of the active channel if there are multiple inputs and
+      // there is new switch message
+      if (is_new_switch_message) {
+        // Check if the channel name is a key of the map. If it is, we update
+        // the active channel name and clear switch_sub_'s message. If it is
+        // not we do not update the active channel name.
+        if (name_to_input_sub_map_.count(switch_sub_->message().channel) == 1) {
+          active_channel_ = switch_sub_->message().channel;
+        } else {
+          std::cout << switch_sub_->message().channel << " doesn't exist\n";
+        }
+
+        // Clear messages in the switch channel
+        name_to_input_sub_map_.at(active_channel_).clear();
+        switch_sub_->clear();
+
+        // Clear messages in the new input channel if we just switched input
+        // channel in the current loop
+        if (previous_active_channel_name.compare(active_channel_) != 0) {
+          name_to_input_sub_map_.at(active_channel_).clear();
```

Comment:
Doesn't seem like there's a need for this second `.clear()` call here. I do think it's always OK to clear the new active channel (line 224), so probably just need to remove 229-231.

Reply:
Line 224 was unintentional, and I've removed it. So the idea is that we only clear the input channel after 
1. we read the input messgae
2. we just switch to a new input channel